src/networking.c

```diff
@@ -2292,8 +2291,12 @@ int processMultibulkBuffer(client *c) {
             }
 
             c->qb_pos = newline-c->querybuf+2;
-            if (ll >= PROTO_MBULK_BIG_ARG) {
-                /* If we are going to read a large object from network
+            if (!(c->flags & CLIENT_MASTER) && ll >= PROTO_MBULK_BIG_ARG) {
```

Comment:
@sundb @yoav-steinberg do you remember something from our observations of #9003 about the impact of this BIG_ARG feature?
IIRC it doesn't have a huge impact on performance, but i can't locate any benchmark we did on it.

Reply:
AFAIK, the only change to BIG_ARG in #9003 is to avoid greedy expansion of querybuf, I forget if I benchmarked it, let me do a regression test.

Reply:
Three tests were performed (before #9003, after #9001, this PR).
It does not appear that performance has changed.

benchmark  command:

```sh
dd if=/dev/zero of=bigfile bs=1M count=32
./src/redis-benchmark -n 100000 -x hset a a < bigfile
```

Best results after multiple tests:
```sh
# optimize-pending_querybuf
  throughput summary: 8152.62 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
        3.178     1.568     3.103     3.991     4.287     6.047

# before #9003
  throughput summary: 8154.61 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
        3.181     1.352     3.103     3.967     4.303     9.311

# after 9100
  throughput summary: 8138.02 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
        3.181     1.424     3.103     3.951     4.263     5.567
```